### ISSUE TITLE
TypeScript Builds Fail with CmsComponent

### ISSUE THREAD

**Issue by garrypolley** [Crownpeak/DXM-React-SDK#8]
I'm using NextJS and TypeScript to attempt to build out our CMS components. 

With a vanilla NextJS project -- built on this one -- https://github.com/garrypolley/with-typescript-app

I get an error when I attempt to use _any_ TypeScript based CMS Component. I'm building off of master (technically in a fork here because I wanted to npm install from github https://github.com/garrypolley/DXM-React-SDK/commits/master)

My base page `HomePage.tsx`:

```tsx
import React from 'react';
import {CmsDynamicPage} from 'crownpeak-dxm-react-sdk';
import Footer from "../components/administrative/footer";



export default class HomePage extends CmsDynamicPage {
    render() {
        super.render();
        return (
            <div>
                <div id="page" className="hfeed site grid-container container grid-parent">
                    <div id="content" className="site-content">
                        <div data-elementor-type="wp-post" data-elementor-id="14804"
                             className="elementor elementor-14804"
                             data-elementor-settings="[]">
                            <div className="elementor-inner">
                                <div className="elementor-section-wrap">
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                <Footer/>
            </div>
        )
    }
}
```

And a footer of `Footer.tsx`: 

```tsx
import React from 'react';
import {CmsComponent} from 'crownpeak-dxm-react-sdk';

export default class Footer extends CmsComponent {
    render() {
        return (
            <div className="site-footer">
            <p>Content that doesn't matter </p>
           </div>
)
}
}
```

I get the following error when I attempt to build the project with the `npm run build`: 

```
./pages/NextHomePage.tsx:38:18
Type error: 'Footer' cannot be used as a JSX component.
  Its instance type 'Footer' is not a valid JSX element.
    Type 'Footer' is missing the following properties from type 'ElementClass': context, setState, forceUpdate, props, and 2 more.

  36 |                     </div>
  37 |                 </div>
> 38 |                 <Footer/>
     |                  ^
  39 |             </div>
  40 |         )
````

It seems like anything that is inheriting from the CmsComponent is not getting the attributes and properties from `React.Component`

The tsconfig. 

```json
{
  "compilerOptions": {
    "allowJs": true,
    "alwaysStrict": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "lib": ["dom", "es2017"],
    "module": "esnext",
    "moduleResolution": "node",
    "noEmit": true,
    "noFallthroughCasesInSwitch": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "resolveJsonModule": true,
    "skipLibCheck": true,
    "strict": true,
    "target": "esnext"
  },
  "exclude": ["node_modules"],
  "include": ["**/*.ts", "**/*.tsx"]
}
```

**Comment by garrypolley** [comment]
alright so I fixed this a little bit by committing the build files to linked repos. I had to build the sdk core and this react sdk. 

Once built upon latest master of each and having all libraries updated to latest react and react types it seems to build without an error now.

**Comment by garrypolley** [comment]
I have forked this repo and the core SDK to get this working: 

https://github.com/garrypolley/DXM-React-SDK

https://github.com/garrypolley/DXM-SDK-Core

The main changes are in the tsconfig files and the package dependencies.

**Comment by garrypolley** [comment]
I think this is fixed by the recent release of 3.0.
